src/iterated_integrals.jl

```diff
@@ -0,0 +1,237 @@
+    WikJ = false .* ΔW .* ΔW'
+    WikJCommute_oop{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJCommute_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJCommute_iip{typeof(WikJ)}(WikJ)
+end
+
+function fill_WikJGeneral_oop(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    WikJCommute_oop{eltype(ΔW), typeof(WikJ)}(WikJ, m_seq)
+end
+
+function fill_WikJGeneral_iip(ΔW)
+    WikJ = false .* ΔW .* ΔW'
+    WikJ2 = similar(WikJ)
+    WikJ3 = similar(WikJ)
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    vec_ζ = false .* vec(ΔW)
+    vec_η = similar(vec_ζ)
+    Gp1 = zeros(M)
+    Gp2 = similar(Gp1)
+    WikJCommute_iip{eltype(ΔW), typeof(WikJ)}(WikJ, WikJ2, WikJ3, m_seq, vec_ζ, vec_η, Gp1, Gp2)
+end
+
+function get_iterated_I!(dW, Wik::WikJDiagonal_oop)
+    @unpack WikJ = Wik
+    WikJ = 1//2 .* dW .* dW
+    WikJ
+end
+
+function get_iterated_I!(dW, Wik::WikJDiagonal_iip)
+    @unpack WikJ = Wik
+    if typeof(dW) <: Number
+        Wik.WikJ = 1//2 .* dW .^ 2
+    else
+        @.. WikJ = 1//2*dW^2
+        @.. Wik.WikJ = WikJ
+    end
+    return nothing
+end
+
+function get_iterated_I!(dW, Wik::WikJCommute_oop)
+    @unpack WikJ = Wik
+    WikJ = 1//2 .* vec(dW) .* vec(dW)'
+    WikJ
+end
+
+function get_iterated_I!(dW, Wik::WikJCommute_iip)
+    @unpack WikJ = Wik
+    mul!(WikJ,vec(dW),vec(dW)')
+    @.. WikJ *= 1//2
+
+    @.. Wik.WikJ = WikJ
+    return nothing
+end
+
+function get_iterated_I!(dW, Wik::WikJGeneral_oop)
+    @unpack WikJ, m_seq = Wik
+
+    m      = length(dW)
+    M      = m*(m-1)/2
+    sum_dW² = dW'*dW
+
+    WikJ = dW*dW'
+    Gp1 = randn(M)
+    α = sqrt(1 + sum_dW²/dt)
+    Gp2 = Gp1/(sqrt(2)*(1+α)*dt)
+
+    #operator (Iₘ² - Pₘ)Kₘᵀ
+    for i in 1:M
+        WikJ2[m_seq[i,1], m_seq[i,2]] = Gp2[i]
+        WikJ2[m_seq[i,2], m_seq[i,1]] = -Gp2[i]
+    end
+
+    #operator (Iₘ X W*Wᵀ)
+    WikJ2 = WikJ*WikJ2
+
+    #operator Kₘ(Iₘ² - Pₘ)
+    WikJ2 = WikJ2 - WikJ2'
+    for i in 1:M
+        Gp2[i] = WikJ2[m_seq[i,1], m_seq[i,2]]
+    end
+    Gp = Gp/sqrt(2) + Gp2
+
+    #operator (Iₘ² - Pₘ)Kₘᵀ
+    for i in 1:M
+        WikJ2[m_seq[i,1], m_seq[i,2]] = Gp[i]
+        WikJ2[m_seq[i,2], m_seq[i,1]] = -Gp[i]
+    end
+
+    WikJ *= 1//2
+    a2ₚ = (π^2)/6
+    p = Int(floor((1/π)*sqrt(M/(24*dt))*sqrt(m + 4*sum_dW²/dt) + 1))
+    for i in 1:p
+        a2ₚ -= (1/i^2)
+        var = sqrt(dt/(2*π*i))
+        vec_ζ = randn(m)*var
+        vec_η = randn(m)*var
+        WikJ += (vec_ζ*vec_η' - vec_η*vec_ζ')
+        Aₚ -= (2/sqrt(π*i))*vec_ζ
+    end
+
+    WikJ -= 1//2*(dW*Aₚ' - Aₚ*dW')
+    WikJ += (sqrt(a2ₚ)*dt/π)*WikJ2
```

Comment:
why is there a dt here?

Reply:
![image](https://user-images.githubusercontent.com/26359940/62457606-d3553700-b798-11e9-9ddc-1178255a3ee2.png)
This is the scalar `Delta_t`/`2*pi`*`ap^0.5`. But I skipped the division by 2 due to cancellation